rsocket-core/src/main/java/io/rsocket/frame/ExtensionFrameFlyweight.java

```diff
@@ -0,0 +1,34 @@
+package io.rsocket.frame;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+public class ExtensionFrameFlyweight {
+  private ExtensionFrameFlyweight() {}
+
+  public static ByteBuf encode(
+      ByteBufAllocator allocator, int streamId, int extendedType, ByteBuf data) {
+    int flags = FrameHeaderFlyweight.FLAGS_I | FrameHeaderFlyweight.FLAGS_M;
```

Comment:
Metadata is not written - flag should not be set (or support writing both data and metadata - and update logic accordingly)